tests/test_unit.py

```diff
@@ -54,3 +55,64 @@ def test_repeating_parameters(add_duplicate):
         ]
     else:
         assert instrument.get_par_by_name("duplicate-name") == p1
+
+
+@pytest.mark.parametrize("value",  [25, 25., 25.64547871216879451687311211245117852145229614585985498212321, "aaaa"])
+def test_float_defaults(value):
+    if type(value) == str:
+        with pytest.raises(RuntimeError):
+            Float(
+                value=value,
+                name="p_float"
+            )
+    else:
+        p_float = Float(
+            value=value,
+            name="p_float"
+        )
+
+        assert p_float.get_value_in_default_format(p_float.value) == p_float.value
+        # assign an int value, that then should be converted to float
+        p_float.value = value
+        assert p_float.get_value_in_default_format(p_float.value) == float(value)
+        assert type(p_float.value) == float
+
+
+@pytest.mark.parametrize("e_units", ['eV', 'W', '', None])
+def test_spectral_boundaries_defaults(e_units):
+    # test with a not allowed unit
+    if e_units == 'W':
+        with pytest.raises(RuntimeError):
+            SpectralBoundary(
+                value=10.,
+                name="p_spectral_boundary",
+                E_units=e_units,
+            )
+    else:
+        p_spectral_boundary = SpectralBoundary(
+            value=10.,
+            name="p_spectral_boundary",
+            E_units=e_units,
+        )
+
+        assert p_spectral_boundary.get_value_in_default_format(p_spectral_boundary.value) == p_spectral_boundary.value
+        assert p_spectral_boundary.get_value_in_default_format(p_spectral_boundary.value) == float(10.)
+        assert type(p_spectral_boundary.value) == float
+
+
+@pytest.mark.parametrize("value",  [25, 25., 25.64547871216879451687311211245117852145229614585985498212321, "aaaa"])
+def test_integer_defaults(value):
+    if not isinstance(value, int):
+        with pytest.raises(RuntimeError):
+            Integer(
+                value=value,
+                name="p_integer"
+            )
+    else:
+        p_integer = Integer(
+            value=value,
+            name="p_integer"
+        )
+        assert p_integer.value == int(value)
+        assert p_integer.get_value_in_default_format(p_integer.value) == int(value)
```

Comment:
casting to `int` here is mostly unnecessary since in this branch it's already int